projects/storefrontlib/src/cms-components/cart/add-to-cart/add-to-cart.component.ts

```diff
@@ -43,14 +55,41 @@ export class AddToCartComponent implements OnInit, OnDestroy {
     quantity: new FormControl(1),
   });
 
+
+
+  constructor(
+    modalService: ModalService,
+    currentProductService: CurrentProductService,
+    cd: ChangeDetectorRef,
+    activeCartService: ActiveCartService,
+    componentData: CmsComponentData<model>
+  );
+
+  /**
+   * @deprecated since 3.4
+   */
+  constructor(
+    modalService: ModalService,
+    currentProductService: CurrentProductService,
+    cd: ChangeDetectorRef,
+    activeCartService: ActiveCartService,
+  );
+
   constructor(
     protected modalService: ModalService,
     protected currentProductService: CurrentProductService,
     private cd: ChangeDetectorRef,
-    protected activeCartService: ActiveCartService
+    protected activeCartService: ActiveCartService,
+    @Optional() protected componentData?: CmsComponentData<model>
   ) {}
 
   ngOnInit() {
+    this.componentData?.data$.subscribe((data: model) => {
+      if (data.inventoryDisplay && data.inventoryDisplay == 'true') {
+        this.showInventory = true;
+      }
+    });
+
     if (this.product) {
       this.productCode = this.product.code;
```

Comment:
productCode type should be set as 'string | undefined' instead of just 'string'

Reply:
done